### PR TITLE
Login/logout 

### PR DESCRIPTION
**Pull request by luciademoja**
PT Story: https://www.pivotaltracker.com/story/show/129046689

Changes proposed in this pull request:

_When user logged in:_
- Changes login to logout
- Greets user in header
- Hides "Register your restaurant" from footer
- Hides "Sign up" from header 

What I have learned working on this feature:
How to test for the current_user to be logged in and use it to change the view.

Screenshots:
skarmavbild 2016-08-28 kl 09 04 33

Ready for review!

### CODE REVIEW

File: app/views/layouts/partials/_header.html.haml [hunk 1]
```diff
@@ -8,6 +8,12 @@
       %li
         = link_to 'Home', root_path
       %li
-        = link_to 'Login', new_user_session_path
-      %li
         = link_to 'Sign Up', new_user_registration_path
+      - if current_user
```
Comment (tochman) [inline app/views/layouts/partials/_header.html.haml:10]: Is the "Sign Up" link visible when a user is logged in?

**Comment by tochman** [comment]
I would also like to see that the link in the footer saying "Register your restaurant" is hidden if a user is logged in.

**Comment by tochman** [comment]
As to the screenshot, I would prefer the styled interface. You can generate it with capybara screenshot.

File: app/views/layouts/partials/_header.html.haml (hunk 1 above)
Comment (luciademoja) [reply app/views/layouts/partials/_header.html.haml:10]: @tochman not anymore ;)

**Comment by diraulo** [comment]
@luciademoja merge conflicts here

**Comment by luciademoja** [comment]
@diraulo they came up after merging the dashboard PR, I'll check this now 👍

File: app/views/layouts/partials/_header.html.haml [hunk 2]
```diff
@@ -7,10 +7,22 @@
...
+          = link_to 'Logout', destroy_user_session_path, method: :delete
+      - else
+        %li
+          = link_to 'Login', new_user_session_path
         %li
-          = link_to 'Dashboard', dashboard_path(current_user.restaurants.first)
+          = link_to 'Sign Up', new_user_registration_path
+
+
+          -# - elsif current_user && current_user.restaurants.any?
```
Comment (diraulo) [inline app/views/layouts/partials/_header.html.haml]: @luciademoja Will this dead code be used later on? If I'd remove it

Comment (tochman) [reply app/views/layouts/partials/_header.html.haml]: That line was used to condition the dashboard link. Not sure if that's still the case.

Comment (diraulo) [reply app/views/layouts/partials/_header.html.haml]: @tochman I'm talking about the block of commented code from this line, not just the line. If it's waiting for another feature to be implemented then we can merge this PR.

Comment (tochman) [reply app/views/layouts/partials/_header.html.haml]: Okay. I don't see any reason to keep that block.

**Comment by luciademoja** [comment]
@MikaelFeher @tochman @diraulo this should be ready to merge now, all tests are passing

**Pull request merged**